### PR TITLE
Add population rank to populated features in the places layer.

### PR DESCRIPTION
**Pull request by zerebubuth**
Adjust collision rank based on population rank.

Connects to #988.

### CODE REVIEW

File: docs/layers.md [hunk 1]
```diff
@@ -732,6 +732,7 @@ Places with `kind` values of `continent`, `country`, with others added starting
 * `id`: The `osm_id` from OpenStreetMap or Natural Earth id
 * `kind`: normalized values between OpenStreetMap and Natural Earth
 * `population`: population integer values from OpenStreetMap or Natural Earth's maximum population value.
+* `population_rank`: A value from 18 down to 0, indicating how large the population is on a particular place. A larger value indicates a bigger population.
```
Comment (nvkelso) [inline docs/layers.md]: Please add full documentation for the class ranges in a new sub-section here (we do this for various kind_detail enumerations, for example).

Comment (nvkelso) [reply docs/layers.md]: https://github.com/tilezen/vector-datasource/issues/988#issuecomment-453262892

```
18: Over 1b
17: 100m to 1b
16: 50m to 100m
15: 20m to 50m
14: 10m to 20m
13: 5m to 10m
12: 1m to 5m
11: 500k to 1m
10: 200k to 500k
9: 100k to 200k
8: 50k to 100k
7: 20k to 50k
6: 10k to 20k
5: 5k to 10k
4: 2k to 5k
3: 1k to 2k
2: 200 to 1k
1: Less than 200
0: Locale (no population) / places that only are visible at 50k scale or larger (but not neighbourhoods)
```

File: spreadsheets/collision_rank.yaml [hunk 2]
```diff
@@ -8,7 +8,25 @@
 - {$layer: 'water', kind: 'ocean'}
 - {$layer: 'places', kind: 'country'}
 # reserved for population_rank
-- _reserved: {count: 19}
+- {$layer: 'places', kind: 'locality', population_rank: 18}
```
Comment (nvkelso) [inline spreadsheets/collision_rank.yaml]: The scene file also adjusts the priority based on `country_capital` and `region_capital` boolean properties, where country is best, then region, then last general locality of that same population rank. Let's adjust this logic here, too (will increase the range past the reserved gap, so some other tests may now fail and need an update).

Comment (nvkelso) [reply spreadsheets/collision_rank.yaml]: Something like:

```
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 18}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 18}
- {$layer: 'places', kind: 'locality', population_rank: 18}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 17}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 17}
- {$layer: 'places', kind: 'locality', population_rank: 17}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 16}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 16}
- {$layer: 'places', kind: 'locality', population_rank: 16}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 15}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 15}
- {$layer: 'places', kind: 'locality', population_rank: 15}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 14}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 14}
- {$layer: 'places', kind: 'locality', population_rank: 14}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 13}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 13}
- {$layer: 'places', kind: 'locality', population_rank: 13}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 12}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 12}
- {$layer: 'places', kind: 'locality', population_rank: 12}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 11}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 11}
- {$layer: 'places', kind: 'locality', population_rank: 11}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 10}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 10}
- {$layer: 'places', kind: 'locality', population_rank: 10}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 9}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 9}
- {$layer: 'places', kind: 'locality', population_rank: 9}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 8}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 8}
- {$layer: 'places', kind: 'locality', population_rank: 8}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 7}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 7}
- {$layer: 'places', kind: 'locality', population_rank: 7}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 6}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 6}
- {$layer: 'places', kind: 'locality', population_rank: 6}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 5}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 5}
- {$layer: 'places', kind: 'locality', population_rank: 5}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 4}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 4}
- {$layer: 'places', kind: 'locality', population_rank: 4}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 3}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 3}
- {$layer: 'places', kind: 'locality', population_rank: 3}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 2}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 2}
- {$layer: 'places', kind: 'locality', population_rank: 2}
- {$layer: 'places', kind: 'locality', country_capital: True, population_rank: 1}
- {$layer: 'places', kind: 'locality', region_capital: True, population_rank: 1}
- {$layer: 'places', kind: 'locality', population_rank: 1}
- {$layer: 'places', kind: 'locality', country_capital: True}
- {$layer: 'places', kind: 'locality', region_capital: True}
- {$layer: 'places', kind: 'locality'}
```

File: spreadsheets/collision_rank.yaml [hunk 3]
```diff
@@ -8,7 +8,25 @@
 - {$layer: 'water', kind: 'ocean'}
 - {$layer: 'places', kind: 'country'}
```
Comment (nvkelso) [inline spreadsheets/collision_rank.yaml]: We also do some population grading of `country` features client side, so let's also include that in this PR, please.

Something like:

```
- {$layer: 'places', kind: 'country', population_rank: 18}
- {$layer: 'places', kind: 'country', population_rank: 17}
- {$layer: 'places', kind: 'country', population_rank: 16}
- {$layer: 'places', kind: 'country', population_rank: 15}
- {$layer: 'places', kind: 'country', population_rank: 14}
- {$layer: 'places', kind: 'country', population_rank: 13}
- {$layer: 'places', kind: 'country', population_rank: 12}
- {$layer: 'places', kind: 'country', population_rank: 11}
- {$layer: 'places', kind: 'country', population_rank: 10}
- {$layer: 'places', kind: 'country', population_rank: 9}
- {$layer: 'places', kind: 'country', population_rank: 8}
- {$layer: 'places', kind: 'country', population_rank: 7}
- {$layer: 'places', kind: 'country', population_rank: 6}
- {$layer: 'places', kind: 'country', population_rank: 5}
- {$layer: 'places', kind: 'country', population_rank: 4}
- {$layer: 'places', kind: 'country', population_rank: 3}
- {$layer: 'places', kind: 'country', population_rank: 2}
- {$layer: 'places', kind: 'country', population_rank: 1}
- {$layer: 'places', kind: 'country'}
```

File: spreadsheets/collision_rank.yaml (hunk 2 above)
Comment (nvkelso) [reply spreadsheets/collision_rank.yaml]: Adjust to taste for the line:

> - _reserved: {count: 1}

File: spreadsheets/collision_rank.yaml (hunk 3 above)
Comment (nvkelso) [reply spreadsheets/collision_rank.yaml]: Please also do for `region` (lower down):

```
- {$layer: 'places', kind: 'region', population_rank: 18}
- {$layer: 'places', kind: 'region', population_rank: 17}
- {$layer: 'places', kind: 'region', population_rank: 16}
- {$layer: 'places', kind: 'region', population_rank: 15}
- {$layer: 'places', kind: 'region', population_rank: 14}
- {$layer: 'places', kind: 'region', population_rank: 13}
- {$layer: 'places', kind: 'region', population_rank: 12}
- {$layer: 'places', kind: 'region', population_rank: 11}
- {$layer: 'places', kind: 'region', population_rank: 10}
- {$layer: 'places', kind: 'region', population_rank: 9}
- {$layer: 'places', kind: 'region', population_rank: 8}
- {$layer: 'places', kind: 'region', population_rank: 7}
- {$layer: 'places', kind: 'region', population_rank: 6}
- {$layer: 'places', kind: 'region', population_rank: 5}
- {$layer: 'places', kind: 'region', population_rank: 4}
- {$layer: 'places', kind: 'region', population_rank: 3}
- {$layer: 'places', kind: 'region', population_rank: 2}
- {$layer: 'places', kind: 'region', population_rank: 1}
- {$layer: 'places', kind: 'region'}
```

Comment (zerebubuth) [reply spreadsheets/collision_rank.yaml]: Fixed in 646eec7.

File: spreadsheets/collision_rank.yaml (hunk 2 above)
Comment (zerebubuth) [reply spreadsheets/collision_rank.yaml]: Fixed in 5e683a7.

File: docs/layers.md (hunk 1 above)
Comment (zerebubuth) [reply docs/layers.md]: Added in 06b65aa.

**Pull request merged**